xwiki-rendering-macros/xwiki-rendering-macro-toc/src/main/java/org/xwiki/rendering/macro/toc/TocEntryDecorator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.macro.toc;
+
+import java.util.List;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.rendering.block.HeaderBlock;
+
+/**
+ * Decorator taking an entry and manipulating the list of blocks used to render the header.
```

Comment:
You cannot document a decorator saying it's a decorator ;) What does it do?

A decorator seems to adds methods/features, see https://refactoring.guru/design-patterns/decorator. Is it the right name?

Question: why do you need to generate several blocks? Also why are generated blocks not `HeaderBlock`s?

Also, I think it would help if the javadoc was listing some examples of decorators, to understand better what it does. Examples go a long way to explain stuff.

Reply:
> You cannot document a decorator saying it's a decorator ;) What does it do?
>
> A decorator seems to adds methods/features, see https://refactoring.guru/design-patterns/decorator. Is it the right name?

Indeed that's not a good idea to reuse the name of a know pattern.

WDYT of `Enhancer` or `Extension`?

> Question: why do you need to generate several blocks? Also why are generated blocks not HeaderBlocks?

The list of blocks is the content of the `LinkBlock` used for a ToC entry (linking to the corresponding header).

> Also, I think it would help if the javadoc was listing some examples of decorators, to understand better what it does. Examples go a long way to explain stuff.

+1